lib/shape.ml

```diff
@@ -3,55 +3,57 @@ open Graphics
 type point = { x : int; y : int }
 type rectangle = { c : point; length : int; width : int }
 type circle = { c : point; radius : int }
-type ellipse = {c : point; rx : int; ry: int}
+type ellipse = { c : point; rx : int; ry: int }
 type shape = Circle of circle | Rectangle of rectangle | Ellipse of ellipse
 type shapes = shape list
 
 let dimensions = ref {x = 500; y = 500}
 let set_dimensions x y = 
-  dimensions := {x;y} 
+  dimensions := {x; y}
 
-let canvas_mid = { x = (!dimensions.x / 2); y = (!dimensions.y / 2)}
+let canvas_center = ref {x = 0; y = 0}
 
 let axes_flag = ref false
 let draw_axes flag = 
   axes_flag := flag
 
 let render_axes () = 
   set_color (rgb 192 192 192);
-  moveto (size_x () / 2) 0;
-  lineto (size_x () / 2) (size_y ());
-  moveto 0 (size_y () / 2);
-  lineto (size_x ()) (size_y () / 2)
+  moveto (!dimensions.x / 2) 0;
+  lineto (!dimensions.x / 2) (!dimensions.y);
+  moveto 0 (!dimensions.y / 2);
+  lineto (!dimensions.x) (!dimensions.y / 2)
 
 let render_shape s =
   match s with
-  | Circle circle -> draw_circle circle.c.x circle.c.y circle.radius
+  | Circle circle -> draw_circle (circle.c.x + !canvas_center.x) (-circle.c.y + !canvas_center.y) circle.radius
```

Comment:
Let's make a function called `denormalise x y` that will take some `x` and `y` from a `(0, 0)` system and convert it to the graphics library's canvas.

Also, this is not correct. `x` would change to `x + canvas_mid.x` and similarly for `y`.